### ISSUE TITLE
Staging Review finding: Focus lost in page transition

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#108125]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** Disability Benefits Experience Team 2
**Product:** Disability Comp Application (21-526 EZ)
**Feature:** Updated 0781 Flow

## Findings details
**VA.gov Experience Standard - issue:** [User experiences predictable interactions with components and patterns.](https://design.va.gov/about/experience-standards/#user-can-identify-an-element-and-its-state)
**VA.gov Experience Standard - category:** [Usability](https://design.va.gov/about/experience-standards/#category-usability)
**Launch-blocking:** Yes
**Design System review:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

When moving from `/event-details` to `/event-report`, keyboard focus is lost. Besides inconveniencing keyboard users, losing focus here also means that nothing is announced to screen reader users on that page transition --- meaning they will have to guess whether or not they've successfully moved to the next page or if something may have gone wrong and they need to try again. It's a potentially confusing and disorienting moment in the flow.

## Recommended action

Focus management is great everywhere else in this flow, so I'm guessing this was an oversight and an accident. But it may be worth double-checking focus management throughout the form just to verify / make sure I didn't miss any either.

## References

- **Accessibility Defect Severity:** [1: Critical. Must be fixed before launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-1)
- **WCAG Success Criteria:** N/A
- **Modality:** Screen reader
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A

---

### Next Steps for the VFS Team

_How did this touchpoint go? [Give feedback on the Collaboration Cycle](https://dj540s05.optimalworkshop.com/questions/f54355271969f160ef41182ba9ceb61c) at any time._
- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by lisacapaccioli** [comment]
Here are my test results of the fix from staging. It works better than before, but it is doing something strange that it adds "article" on the page focus. I am not sure what that is, but if you watch the video that might help. (I don't see "article" showing up on other page focus). 

https://github.com/user-attachments/assets/14909575-69e4-44b5-a8d5-862c373718b6

**Comment by freeheeling** [comment]
@kylesoskin, @briandeconinck In response to Lisa's noted observations above: The screen reader is likely exposing the heading element's semantic region prior to reading its contents. "Article" appears to be coming from an internal `<h2>`, inside a Shadow DOM of `<va-segmented-progress-bar>`, where the element cannot be reliably focused. The screen reader announces the progress bar's heading because that is where focus is directed. Focusing on the progress bar was intentional to maintain consistency with all other pages within 526. By default, focus is otherwise directed to the page heading as part of the Array Builder Pattern. As a workaround, I tried to skip focusing the progress bar by focusing a hidden, custom `<h2>` with a `sr-only` class after the progress bar, but this and other attempts to circumvent the "article" announcement were unsuccessful. Therefore, unless there are any objections concerning meeting accessibility criteria, I'm inclined to not continue trying to refine the improved screen reading. Please provide me with your thoughts, or suggestions if you have any. Thanks.

**Comment by briandeconinck** [comment]
@freeheeling and @lisacapaccioli I just re-tested and am verifying that focus goes to the progress bar, and I think the current state is sufficient. Thank you!

**Comment by freeheeling** [comment]
Thank you for confirming, @briandeconinck.
